### ISSUE TITLE
Unsatisfactory results

### ISSUE THREAD

**Issue by lulujianjie** [Li-Chongyi/Zero-DCE#4]
input:

![timg-input](https://user-images.githubusercontent.com/9878804/86428664-e57c8400-bd1f-11ea-865d-f4cac664a6a6.jpeg)

output:

![timg-result](https://user-images.githubusercontent.com/9878804/86428673-ea413800-bd1f-11ea-90c2-09a9351df156.jpeg)

**Comment by Li-Chongyi** [comment]
Such an image with obvious noise challenges all low-light image enhancement methods. Zero-DCE can only enhance low-light image without obvious artifact or noise. Unfortunately, it cannot handle artifacts, blocking, and noise well.

发件人：JackLu <notifications@github.com>
发送日期：2020-07-03 11:26:17
收件人：Li-Chongyi/Zero-DCE <Zero-DCE@noreply.github.com>
抄送人：Subscribed <subscribed@noreply.github.com>
主题：[Li-Chongyi/Zero-DCE] Unsatisfactory results (#4)

input:

output:

**Comment by lulujianjie** [comment]
In fact, I have tried many images without obvious noise. And I compared the results generated from your model and from MEITU (only adjust in low level, not using deep model).

One of the examples shows  as following:
input
<img src="https://s1.ax1x.com/2020/07/03/NO2Lm4.jpg">

your DCE output
<img src="https://s1.ax1x.com/2020/07/03/NO2bXF.jpg">

MEITU output(only add 30 brightness)
<img src="https://s1.ax1x.com/2020/07/03/NO2O0J.jpg">

My question is **if using the deep model to learn a simple brightness transformation will be easy to be overfiting**. Although your model is not very large, it also has excessive capacity to fit a simple transformation.

**Comment by Li-Chongyi** [comment]
MEITU is commercial software that contains a serial of SP operations, not just light enhancement. Moreover, I guess that it operates on raw data (not sure). So it is unfair to compare it with. There is no overfitting because Zero-DCE is an optimization-based method without ground truth. If the Zero-DCE is used in the product, it needs some modifications such as coupled with denoising and removing color band, etc. Anyway, thanks for providing the resutls.

发件人：JackLu <notifications@github.com>
发送日期：2020-07-03 12:43:11
收件人：Li-Chongyi/Zero-DCE <Zero-DCE@noreply.github.com>
抄送人：Chongyi Li <lichongyi25@gmail.com>,Comment <comment@noreply.github.com>
主题：Re: [Li-Chongyi/Zero-DCE] Unsatisfactory results (#4)

In fact, I have tried many images without obvious noise. And I compared the results generated from your model and from MEITU (only adjust in low level, not using deep model).
One of the examples shows as following:
 input

your DCE output

MEITU output(only add 30 brightness)

My question is if using the deep model to learn a simple brightness transformation will be easy to be overfiting. Although your model is not very large, it also has excessive capacity to fit a simple transformation.

**Comment by maldeer-Winlab** [comment]
Hi, @lulujianjie 
This is Murtadha Aldeer, a PhD candidate at Rutgers University, USA. I apologize as I am posting something irrelevant to this topic but I sent you many emails and every time I get delivery failure. 
I have questions regarding using your code you developed for "Robust Single Accelerometer-Based Activity Recognition Using Modified Recurrence Plot" as I am doing some related work. 

I appreciate it if you provide me with a working email o contact you. 
My email address is maldeer@winlab.rutgers.edu

Appreciating your reply

Thank you

**Comment by lulujianjie** [comment]
> Hi, @lulujianjie
> This is Murtadha Aldeer, a PhD candidate at Rutgers University, USA. I apologize as I am posting something irrelevant to this topic but I sent you many emails and every time I get delivery failure.
> I have questions regarding using your code you developed for "Robust Single Accelerometer-Based Activity Recognition Using Modified Recurrence Plot" as I am doing some related work.
> 
> I appreciate it if you provide me with a working email to contact you.
> My email address is [maldeer@winlab.rutgers.edu](mailto:maldeer@winlab.rutgers.edu)
> 
> Appreciating your reply
> 
> Thank you



> Hi, @lulujianjie
> This is Murtadha Aldeer, a PhD candidate at Rutgers University, USA. I apologize as I am posting something irrelevant to this topic but I sent you many emails and every time I get delivery failure.
> I have questions regarding using your code you developed for "Robust Single Accelerometer-Based Activity Recognition Using Modified Recurrence Plot" as I am doing some related work.
> 
> I appreciate it if you provide me with a working email to contact you.
> My email address is [maldeer@winlab.rutgers.edu](mailto:maldeer@winlab.rutgers.edu)
> 
> Appreciating your reply
> 
> Thank you

Ok, I will send an email to you.

**Comment by lulujianjie** [comment]
> MEITU is commercial software that contains a serial of SP operations, not just light enhancement. Moreover, I guess that it operates on raw data (not sure). So it is unfair to compare it with. There is no overfitting because Zero-DCE is an optimization-based method without ground truth. If the Zero-DCE is used in the product, it needs some modifications such as coupled with denoising and removing color band, etc. Anyway, thanks for providing the resutls. 发件人：JackLu <notifications@github.com> 发送日期：2020-07-03 12:43:11 收件人：Li-Chongyi/Zero-DCE <Zero-DCE@noreply.github.com> 抄送人：Chongyi Li <lichongyi25@gmail.com>,Comment <comment@noreply.github.com> 主题：Re: [Li-Chongyi/Zero-DCE] Unsatisfactory results (#4) In fact, I have tried many images without obvious noise. And I compared the results generated from your model and from MEITU (only adjust in low level, not using deep model). One of the examples shows as following: input your DCE output MEITU output(only add 30 brightness) My question is if using the deep model to learn a simple brightness transformation will be easy to be overfiting. Although your model is not very large, it also has excessive capacity to fit a simple transformation. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

Thanks for your reply. However, this is not a convincing response. There are a lot of inexplicable noises in the night sky of genenrated result as shown in figures, which are obviously caused by the deep model. As far as I know, illumination enhancement is a very basic operation in Meitu, Lightroom, etc.

**Comment by bsun0802** [comment]
@lulujianjie 
Hi, I have a hard time reproducing the astonishing well results as in the ZeroDCE paper as well. However, I should say this non-reference model training idea is cool. 

Based on what described in the ZeroDCE paper, I have [my own implementation of ZeroDCE](https://github.com/bsun0802/Zero-DCE). I took a significant amount try/modify it, including the loss function and parameter setting. 

I encounter problems like bright/white area getting darken, and borders getting messed. But I never encounter such 'noisy point' artifact. Would you mind try my implementation(https://github.com/bsun0802/Zero-DCE) and my pre-trained model?

**Comment by lulujianjie** [comment]
> @lulujianjie
> Hi, I have a hard time reproducing the astonishing well results as in the ZeroDCE paper as well. However, I should say this non-reference model training idea is cool.
> 
> Based on what described in the ZeroDCE paper, I have [my own implementation of ZeroDCE](https://github.com/bsun0802/Zero-DCE). I took a significant amount try/modify it, including the loss function and parameter setting.
> 
> I encounter problems like bright/white area getting darken, and borders getting messed. But I never encounter such 'noisy point' artifact. Would you mind tring my implementation(https://github.com/bsun0802/Zero-DCE) and my pre-trained model?

You may try the two above inputs, which are raw input images in my experiment.

**Comment by maldeer-Winlab** [comment]
Thank you, I will be waiting for your email

**Comment by Amr-Mustafa** [comment]
@Li-Chongyi You said earlier in this thread that this method needs to be coupled with denoising techniques and removing color band if it is to be used in production, I want to ask if you have any starting points on these subjects ?

**Comment by Li-Chongyi** [comment]
Hi,

Thanks for asking. The company has several methods to enhance lowlight
image and denoising with Zero-DCE coupled with a denoising technique.
However, I cannot provide the details because of the conflict of interest.
Sorry.

Amr Mustafa <notifications@github.com> 于2020年7月29日周三 上午2:54写道：

**Comment by zhuozhongshuo** [comment]
@lulujianjie I get the similar results with you by using the default parameters.  Which pytorch version do you use?  My environment is:
- pytorch 1.8.2
- torchvision 0.9.2
- cuda 10.2
a little different with the recommend env.

**Comment by jiayou23** [comment]
> [@lulujianjie](https://github.com/lulujianjie) I get the similar results with you by using the default parameters. Which pytorch version do you use? My environment is:
> 
> * pytorch 1.8.2
> * torchvision 0.9.2
> * cuda 10.2
>   a little different with the recommend env.
Hi, I'm having a similar problem to this. how you solved it?
